core/src/Cuda/Kokkos_Cuda_Parallel.hpp

```diff
@@ -534,6 +534,20 @@ class ParallelFor<FunctorType, Kokkos::MDRangePolicy<Traits...>, Kokkos::Cuda> {
   const Policy m_rp;
 
  public:
+  template <typename Policy, typename Functor>
+  static int max_tile_size_product(const Policy& pol, const Functor&) {
```

Comment:
Why does it need to be a static method?

Reply:
It avoids needing to construct the DriverType

Reply:
Can it be a free function?

Reply:
This turns out to be a very annoying change to write. Do you feel strongly about this being a free function? I can write it, I just wind up having to spell out a lot of types in a lot of places, and deal with all the goofiness that Reducers bring up (switching on InvalidType, the difficulties in constructing them)

Reply:
I think this is ok. Generally we may wanna add a call to the MDRangePolicy itself which exposes this corresponding to how we deal with it in the TeamPolicy but those could just call through I guess?

Reply:
Duplication is unfortunate but I am not blocking